tests/test_create_flatfield.py

```diff
@@ -0,0 +1,88 @@
+    """
+    corgidrp.track_individual_errors = True # this test uses individual error components
+
+    ###### create simulated data
+    # check that simulated data folder exists, and create if not
+    datadir = os.path.join(os.path.dirname(__file__), "simdata")
+    if not os.path.exists(datadir):
+        os.mkdir(datadir) 
+    mocks.create_simflat_dataset(filedir=datadir)
+    
+    # simulated images to be checked in flat division
+    simdata_filenames=glob.glob(os.path.join(datadir, "sim_flat*.fits"))
+    simflat_dataset=data.Dataset(simdata_filenames)
+    
+    ###### create simulated raster scanned data
+    # check that simulated raster scanned data folder exists, and create if not
+    file_dir = os.path.join(os.path.dirname(__file__), "simdata_rasterscan")
+    data_dir = os.path.join(os.path.dirname(__file__),"test_data/")
+    if not os.path.exists(file_dir):
+        os.mkdir(file_dir) 
+    filenames = glob.glob(os.path.join(data_dir, "med*.fits"))
+    data_set = data.Dataset(filenames)
+    mocks.create_onsky_rasterscans(data_set,filedir=file_dir,planet='neptune',band='1')
+    
+    ###### create flat field
+    flat_filenames = glob.glob(os.path.join(file_dir, "*.fits"))
+    flat_dataset = data.Dataset(flat_filenames)
+    onskyflat_field = detector.create_onsky_flatfield(flat_dataset,planet='neptune',band='1')
+   
+    assert np.nanmean(onskyflat_field.data) == pytest.approx(1, abs=1e-2)
+    
+    
+    calibdir = os.path.join(os.path.dirname(__file__), "testcalib")
+    
+    flat_filename = "sim_onsky_flatfield.fits"
+    if not os.path.exists(calibdir):
+        os.mkdir(calibdir)
+    onskyflat_field.save(filedir=calibdir, filename=flat_filename)
+    
+    ###### perform flat division
+    # load in the flatfield
+    flat_filepath = os.path.join(calibdir, flat_filename)
+    onsky_flatfield = data.FlatField(flat_filepath)
+
+    
+    flatdivided_dataset = l2a_to_l2b.flat_division(simflat_dataset,onsky_flatfield)
+    
+    
+	# perform checks after the flat divison for one of the dataset
+    assert(flat_filename in str(flatdivided_dataset[0].ext_hdr["HISTORY"]))
+	
+	# check the propagated errors for one of the dataset
+    assert flatdivided_dataset[0].err_hdr["Layer_2"] == "FlatField_error"
+    print("mean of all simulated data",np.mean(simflat_dataset.all_data))
+    print("mean of all simulated data error",np.nanmean(simflat_dataset.all_err) )
+    print("mean of all flat divided data:", np.nanmean(flatdivided_dataset.all_data))
+    print("mean of flatfield:", np.nanmean(onsky_flatfield.data))
+    print("mean of flatfield err:", np.nanmean(onsky_flatfield.err))
+    
+    err_flatdiv=np.nanmean(flatdivided_dataset.all_err)
+    err_estimated=np.sqrt(((np.nanmean(onsky_flatfield.data))**2)*(np.nanmean(simflat_dataset.all_err))**2+((np.nanmean(simflat_dataset.all_data))**2)*(np.nanmean(onsky_flatfield.err))**2)
+    print("mean of all flat divided data errors:",err_flatdiv)
```

Comment:
I'm not sure what the goal of these lines are. Should this be an assert statement that checks that the 2 are approximately the same?

Reply:
yeah, I missed adding the asset statement, will fix